### PR TITLE
Contributing guidelines

### PR DESCRIPTION
**Pull request by allienworks**
- added Contributing guidelines
- updated readme

Needs proof-reading and fact-checking! Maybe you have better solutions, I'd love to hear them :)

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,106 @@
+# Contributing guidelines
+
+First of all, thanks for your interest in helping us build Particl! As more coders are joining our cause, we had to create these guidelines to set some standards and keep everyone on the same track. So here are a few rules and tips to get you started faster and make everyone's life a bit easier.
+
+1. [Dev workflow](#dev-workflow)
+    1. [Issues](#1-issues)
+    2. [Push Requests (PRs)](#2-push-requests-prs)
```
Comment (pciavald) [inline CONTRIBUTING.md]: Pull* requests

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+        3. [Submit PR](#23-submit-pr)
+2. [Dev resources](#dev-resources) (Angular, components, layouts, icons)
+
+---
+
+## Dev workflow
+
+Quick overview of the workflow:
+
+1. bug is found / feature is suggested → new Issue
```
Comment (pciavald) [inline CONTRIBUTING.md]: check if it exists before creating an issue (worth reminding)

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+2. new branch is created with code solving the Issue → Push Request initiated
```
Comment (pciavald) [inline CONTRIBUTING.md]: pull*

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+
+When bug is found or new feature/suggestion proposed, [new Issue gets created](https://github.com/particl/partgui/issues). Our repo has predefined Issue template to help you started with that (the more info you provide the better).
+
+- **Always keep the Issue's title short and descriptive!**
+  - Examples:
+    - ✅ _"Pagination in Address book doesn't work (macOS)"_
+    - ❎ _"When you click next page in address book nothing happens and 2 pages are hidden please help!!!1!"_
+- **Include as much related information** in Issue's description as you can (the predefined template will help you with that)
+  - For bugs: what's happening, how to reproduce it, what OS/version are you using, ...
+- **Tag the Issue with Labels** to categorize it (is it a `bug` or just an `enhancement` like a new feature?). If you're not sure which label to use, leave them empty, someone else will do that (better no labels at all than wrong ones)
```
Comment (pciavald) [inline CONTRIBUTING.md]: enhancement is not the same as feature

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+
+After your repo is ready, create new branch.
+
+- **Again, keep branch names short and descriptive**
+- Ideally also tag your branch with `prefix/..` – this helps categorizing all the branches at first glance. Some examples:
+  - `fix/..` - for bug fixing, e.g. `fix/createwallet`
+  - `feature/..` - for new features, e.g. `feature/coldstaking`
+  - `gfx/..` - for graphic/UI/UX tweaks, e.g. `gfx/typography`
+
+When your branch is ready, do your changes and push them to the branch. Be sure to push only the changes related to the connected Issue! Solving multiple Issues is also possible, but it's better to focus on one thing at a time.
```
Comment (pciavald) [inline CONTRIBUTING.md]: fixing several issues in one PR is forbidden for anyone else than you, GFX designer @allienworks

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+
+#### 2.2. Commit your code
+
+- Write meaninful commit messages – don't forget that when your PR is merged, its context is lost. Meaning that once your commits get to `dev` branch, it's harder to say what each commit does just by its name:
```
Comment (pciavald) [inline CONTRIBUTING.md]: we should mention that the code and commits will be human read with attention and PRs will be refused if they are not described correctly

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+  - ❎ Bad commit message: _"fixed button"_
+  - ✅ Good: _"Receive: fixed 'copy address' button"_
+
+#### 2.3. Submit PR
+
+When your code is complete and ready, [submit new Push Request](https://github.com/particl/partgui/compare).
+
+- **Always set your base branch to `dev`** (never to `master` or others!) – `dev` is our main development branch
+- **As always, write short and descriptive title** (you can keep it the same name as your branch)
+- **Describe what has been done in description** – what does this PR solves? We usually list all the changes and even include a screenshot/GIF overview of the final result. This greatly helps the Reviewers and Testers as they know what to focus on.
```
Comment (pciavald) [inline CONTRIBUTING.md]: - picture of the problem
- picture of the solution

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -0,0 +1,106 @@
...
+- **Include references to Issues that this PR solves** – Writing `Fixes #[number]` automatically connects this PR to Issue _[number]_ and when this PR gets merged later, the corresponding Issue gets automatically closed.
```
Comment (pciavald) [inline CONTRIBUTING.md]: writing -> vague. commit message ? PR comment ?

File: README.md [hunk 9]
```diff
@@ -49,12 +47,12 @@ You can directly interact with the daemon ran by the Electron version.
 ## Running
 
 ### Start Electron
-* `npm run start:electron:fast` - disables debug messages for faster startup
```
Comment (pciavald) [inline README.md]: replace all npm commands by yarn / remove the deprecated `--`

**Comment by pciavald** [comment]
can you please also add snyk badge https://github.com/particl/particl-desktop/pull/706/files#diff-04c6e90faac2675aa89e2176d2eec7d8
#706 will be closed as yarn.lock was already added

File: CONTRIBUTING.md (hunk 8 above)
Comment (allienworks) [reply CONTRIBUTING.md]: Good point. Summarizing these all in PR's description is the best IMO.

File: CONTRIBUTING.md (hunk 1 above)
Comment (allienworks) [reply CONTRIBUTING.md]: Hehe, my bad. I always mix the two in "PR" :)

**Comment by pciavald** [comment]
actually don't include snyk i'll merge #706

**Comment by pciavald** [comment]
@allienworks there's still one change :)

**Comment by allienworks** [comment]
@pciavald it's not, fixed already :) dunno why, but GitHub thinks the issue is still there, even if the line was removed by other one.. (or am I just blind? – check the changed files)

**Comment by pciavald** [comment]
yeah perfect @allienworks. can I merge this now or was there a reason for it to be for 1.1.1 ?

**Comment by allienworks** [comment]
Not that I'm aware of. I think we're free to merge; after all this is needed a long time ;)

**Pull request merged**